### PR TITLE
feat: Repair the instrument panel and Echarts components. The indicator name should be filled in

### PR DESCRIPTION
**Pull request by kogel-net**

### CODE REVIEW

File: src/Web/Masa.Tsc.Web.Admin.Rcl/Components/Panel/Chart/ChartPanelConfiguration.razor.cs [hunk 1]
```diff
@@ -35,17 +35,26 @@ protected override void OnInitialized()
...
+        var success = !Value.Metrics.Any(x => string.IsNullOrEmpty(x.Name));
+        if (success)
+        {
+            Value.Metrics.RemoveAll(item => string.IsNullOrEmpty(item.Name));
+            NavigationManager.NavigateToDashboardConfigurationRecord(DashboardId, ServiceName);
+        }
+        else
+        {
+            await PopupService.AlertAsync(T("Name is required"), AlertTypes.Error);
+            return;
```
Comment (doddgu) [inline src/Web/Masa.Tsc.Web.Admin.Rcl/Components/Panel/Chart/ChartPanelConfiguration.razor.cs:49]: Return is meaningless.

File: src/Web/Masa.Tsc.Web.Admin.Rcl/Components/Panel/Chart/ChartPanelConfiguration.razor.cs [hunk 2]
```diff
@@ -35,17 +35,26 @@ protected override void OnInitialized()
         if (Value.Metrics.Any() is false) Value.Metrics.Add(new());
     }
 
-    void NavigateToPanelConfigurationPage()
+    async Task NavigateToPanelConfigurationPage()
```
Comment (doddgu) [inline src/Web/Masa.Tsc.Web.Admin.Rcl/Components/Panel/Chart/ChartPanelConfiguration.razor.cs:38]: NavigateToPanelConfigurationPageAsync

File: src/Web/Masa.Tsc.Web.Admin.Rcl/Components/Panel/Chart/ChartPanelConfiguration.razor.cs [hunk 3]
```diff
@@ -35,17 +35,26 @@ protected override void OnInitialized()
...
+        }
+        else
+        {
+            await PopupService.AlertAsync(T("Name is required"), AlertTypes.Error);
+            return;
+        }
     }
 
-    void Cancel()
+    async Task Cancel()
```
Comment (doddgu) [inline src/Web/Masa.Tsc.Web.Admin.Rcl/Components/Panel/Chart/ChartPanelConfiguration.razor.cs:53]: CancelAsync

**Review by doddgu — CHANGES_REQUESTED** [review]

**Comment by kogel-net** [comment]
Updated

**Comment by 15168440402** [comment]
> Updated

图片

**Comment by kogel-net** [comment]
> > Updated
> 
> 图片

updated

**Comment by 15168440402** [comment]
图片
The prompt information is inconsistent with the TextField Label

**Comment by 15168440402** [comment]
LGTM

**Review by 15168440402 — APPROVED** [review]

File: src/Web/Masa.Tsc.Web.Admin.Rcl/Components/Panel/Chart/ChartPanelConfiguration.razor.cs [hunk 4]
```diff
@@ -35,17 +35,26 @@ protected override void OnInitialized()
...
-        NavigationManager.NavigateToDashboardConfigurationRecord(DashboardId, ServiceName);
+        var success = !Value.Metrics.Any(x => string.IsNullOrEmpty(x.Name));
+        if (success)
+        {
+            Value.Metrics.RemoveAll(item => string.IsNullOrEmpty(item.Name));
+            NavigationManager.NavigateToDashboardConfigurationRecord(DashboardId, ServiceName);
+        }
+        else
+        {
+            await PopupService.AlertAsync(T("Name is required"), AlertTypes.Error);
```
[the remark this answers is not in the document]
Comment (kogel-net) [reply src/Web/Masa.Tsc.Web.Admin.Rcl/Components/Panel/Chart/ChartPanelConfiguration.razor.cs]: updated

**Pull request merged**